Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -1070,3 +1099,121 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Remove Doc Fr
     CHECK(context.pullFilterTotal == 1);
     CHECK(context.pullFilterPurge == 1);
 }
+
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Auto Purge Enabled - Filter Removed Revision SG",
+                 "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    // one collection for now. Will use multiple collection when SG is ready.
+    constexpr size_t collectionCount = 1;
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+        Roses
+    };
+    std::array<C4Collection*, collectionCount> collections =
+        collectionPreamble(collectionSpecs, TestUser, "password");
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+
+    string doc1ID = idPrefix + "doc1";
+    vector<string> chIDs {idPrefix+"a"};
+
+    hasCreatedTestUser = createTestUser(chIDs);
+    REQUIRE(hasCreatedTestUser);
+
+    // Create docs on SG:
+    _authHeader = SGUserAuthHeader;
+    for (size_t i = 0; i < collectionCount; ++i) {
+        sendRemoteRequest("PUT", repl::Options::collectionSpecToPath(collectionSpecs[i]),
+                        doc1ID, addChannelToJSON("{}"_sl, "channels"_sl, chIDs));
+    }
+
+    struct CBContext {
+         int docsEndedTotal = 0;
+         int docsEndedPurge = 0;
+         int pullFilterTotal = 0;
+         int pullFilterPurge = 0;
+         void reset() {
+             docsEndedTotal = 0;
+             docsEndedPurge = 0;
+             pullFilterTotal = 0;
+             pullFilterPurge = 0;
+         }
+     } cbContext;
```

Comment:
This is tons of lines duplicated with other tests, which is a recipe for unreliable maintainability.

Reply:
Do you refer to the above or below?

Reply:
Always above....below does not show up in the snippet above

Reply:
It's just like 4 local variables. Thought to take them out but could not find a meaningful standalone description and there are only 3 or 4 tests may share.